### ISSUE TITLE
Tingle.Extensions.Caching.MongoDB throws sporadically "Collection already exists" when using "CreateIfNotExists = true"

### ISSUE THREAD

**Issue by ThomasTosik** [tinglesoftware/dotnet-extensions#126]
I tested Tingle.Extensions.Caching.MongoDB and when scaling up alot of services instances i get sporadically the error below:

```
MongoDB.Driver.MongoCommandException: Command create failed: Collection already exists. NS: TestCacheDatabase.TestCacheCollection.
   at MongoDB.Driver.Core.WireProtocol.CommandUsingCommandMessageWireProtocol`1.ProcessResponse(ConnectionId connectionId, CommandMessage responseMessage)
   at MongoDB.Driver.Core.WireProtocol.CommandUsingCommandMessageWireProtocol`1.ExecuteAsync(IConnection connection, CancellationToken cancellationToken)
   at MongoDB.Driver.Core.Servers.Server.ServerChannel.ExecuteProtocolAsync[TResult](IWireProtocol`1 protocol, ICoreSession session, CancellationToken cancellationToken)
   at MongoDB.Driver.Core.Operations.CommandOperationBase`1.ExecuteProtocolAsync(IChannelSource channelSource, ICoreSessionHandle session, ReadPreference readPreference, CancellationToken cancellationToken)
   at MongoDB.Driver.Core.Operations.WriteCommandOperation`1.ExecuteAsync(IWriteBinding binding, CancellationToken cancellationToken)
   at MongoDB.Driver.Core.Operations.CreateCollectionOperation.ExecuteAsync(IWriteBinding binding, CancellationToken cancellationToken)
   at MongoDB.Driver.OperationExecutor.ExecuteWriteOperationAsync[TResult](IWriteBinding binding, IWriteOperation`1 operation, CancellationToken cancellationToken)
   at MongoDB.Driver.MongoDatabaseImpl.ExecuteWriteOperationAsync[T](IClientSessionHandle session, IWriteOperation`1 operation, CancellationToken cancellationToken)
   at MongoDB.Driver.MongoDatabaseImpl.CreateCollectionAsync(IClientSessionHandle session, String name, CreateCollectionOptions options, CancellationToken cancellationToken)
   at MongoDB.Driver.MongoDatabaseImpl.UsingImplicitSessionAsync(Func`2 funcAsync, CancellationToken cancellationToken)
   at Tingle.Extensions.Caching.MongoDB.MongoCache.MongoCollectionInitializeAsync()
   at Tingle.Extensions.Caching.MongoDB.MongoCache.ConnectAsync(CancellationToken token)
   at Tingle.Extensions.Caching.MongoDB.MongoCache.GetAsync(String key, CancellationToken token)
   at Microsoft.Extensions.Caching.Distributed.DistributedCacheExtensions.GetStringAsync(IDistributedCache cache, String key, CancellationToken token)
```

When using "CreateIfNotExists = true" i would assume that such erros would not be thrown. It complains that the collection already exists. Which should be still valid (no error) if "CreateIfNotExists = true" is set.

Using:
MongoDB.Driver: 2.19.1
Tingle.Extensions.Caching.MongoDB: 4.1
Net 7
Linux, Docker and Kubernetes

**Comment by ThomasTosik** [comment]
> Is this something that started happening in version 4.1.0 (released on 22 May, 2023)?

I just started with 4.1.0 and have no experience with prior releases.

>Could you possibly be configuring the cache twice in your setup resulting the options being overridden?

When i restart my service it works, but i am confused how this should have an effect.

>Could you try and reproduce with minimal code, maybe offer a public repro of the same?

Thats a little bit hard. You need a cold start with alot of services to achieve this. We have several test clusters where we start cold from scratch with several dozen of containers in parallel. Thats the scenario where its happening.

What i propose is catching this explicit exception and skipping it when "CreateIfNotExists = true" is on.

**Comment by mburumaxwell** [comment]
If I get you right, you are starting lots of replicas at once targeting the same database and collection but the collection is not created prior?

**Comment by mburumaxwell** [comment]
@ThomasTosik this should be fixed in [4.1.1](https://github.com/tinglesoftware/dotnet-extensions/releases/tag/4.1.1). Could you give it a try?
NuGet: https://www.nuget.org/packages/Tingle.Extensions.Caching.MongoDB/4.1.1
